javascript/javascript3/week1/giphy.js

```diff
@@ -0,0 +1,47 @@
+//GIPHY
+document.addEventListener("DOMContentLoaded", giphyInit);
+
+function giphyInit() {
+  const btnSearch = document.querySelector(".btnSearch");
+  btnSearch.addEventListener("click", (ev) => {
+    ev.preventDefault(); // to stop the page reload
+    let numGif = document.querySelector(".num").value;
+    console.log("num:" + numGif);
+
+    let url = `http://api.giphy.com/v1/gifs/search?q=ryan+gosling&api_key=FtR8NF0BZTC8WREmYfUfvQYzxO8uGHBA&limit=${numGif}&q=`;
+
+    let searchInput = document.querySelector(".search").value.trim();
+
+    url = url.concat(searchInput); // concat whatever they tayped inside the form
+    console.log(url);
+    fetch(url)
+      .then((response) => response.json()) //gets the response the second then it handle the data
+      .then((giphy) => {
+        console.log(giphy.meta);
+        console.log(giphy.data);
+        if (searchInput.value == "") {
+          const errorm = document.getElementById("error-m");
+          errorm.innerText = "error type a Gif name first";
```

Comment:
This error doesn't work and doesn't show on your page. Did you want to show this on your page?

Reply:
yes it shows but it doesnt go away until I refresh the page